pilot/pkg/xds/bench_test.go

```diff
@@ -135,6 +141,28 @@ func BenchmarkRouteGeneration(b *testing.B) {
 	}
 }
 
+// Do a quick sanity tests to make sure telemetry v2 filters are applying. This ensures as they
+// update our benchmark doesn't become useless.
+func TestValidateTelemetry(t *testing.T) {
+	s, proxy := setupAndInitializeTest(t, ConfigInput{Name: "telemetry", Services: 1})
+	c, _ := s.Discovery.Generators[v3.ClusterType].Generate(proxy, s.PushContext(), nil, nil)
+	if len(c) == 0 {
+		t.Fatal("Got no clusters!")
+	}
+	for _, r := range c {
+		cls := &cluster.Cluster{}
+		if err := ptypes.UnmarshalAny(r, cls); err != nil {
+			t.Fatal(err)
+		}
+		for _, ff := range cls.Filters {
```

Comment:
why do we only sanity check cluster side of filters?

Reply:
if the cluster is configured everything should be. this is not a telemetry v2 thorough test, just making sure we match an EnvoyFilter and apply it. In the past we have accidentally applied 0 filters - but it would be a strange bug to only match the cluster one and not the listener